velox/type/TimestampUtil.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/base/Exceptions.h"
+
+namespace facebook::velox {
+/// A static class that holds helper functions for TIMESTAMP type.
+class TimestampUtil {
+ public:
+  struct TimestampToStringOptions {
+    enum Precision : int8_t {
+      kMilliseconds = 3,
+      kNanoseconds = 9,
+    } precision = kNanoseconds;
+
+    bool zeroPaddingYear = false;
+    char dateTimeSeparator = 'T';
+    bool dateOnly = false;
```

Comment:
On second thought, let's use a single enum here instead of two booleans, then we won't have invalid state like both being true:
```c++
struct TimestampToStringOptions {
  // ...
  enum Mode : int8_t {
    kFull,
    kDateOnly,
    kTimeOnly,
  } mode = kFull;
};
```

Reply:
Done.